File: src/leetcode/python/RotateArray_189.py
class RotateArray_189(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        l = len(nums)
        if l < 2:
            return
        k %= l
        nums[:] = nums[l - k:] + nums[:l - k]

File: src/leetcode/java/RotateArray_189.java
package leetcode.java;

public class RotateArray_189 {
  public void rotate(int[] nums, int k) {
    if (nums.length < 2) return;
    k %= nums.length;
    reverse(nums, 0, nums.length - 1);
    reverse(nums, 0, k - 1);
    reverse(nums, k, nums.length - 1);
  }

  private void reverse(int[] nums, int i, int j) {
    int tmp;
    while (i < j) {
      tmp = nums[i];
      nums[i++] = nums[j];
      nums[j--] = tmp;
    }
  }
}
